backend/dataall/modules/datasets_base/db/dataset_models.py

```diff
@@ -141,3 +141,23 @@ class Dataset(Resource, Base):
     @classmethod
     def uri(cls):
         return cls.datasetUri
+
+
+class DatasetBucket(Resource, Base):
+    __tablename__ = 'dataset_bucket'
+    datasetUri = Column(String, nullable=False)
+    bucketUri = Column(String, primary_key=True, default=utils.uuid('bucket'))
+    AwsAccountId = Column(String, nullable=False)
+    S3BucketName = Column(String, nullable=False)
+    region = Column(String, default='eu-west-1')
+    partition = Column(String, default='aws')
+    KmsAlias = Column(String, nullable=False)
+    imported = Column(Boolean, default=False)
+    importedKmsKey = Column(Boolean, default=False)
+    userRoleForStorageBucket = query_expression()
+    projectPermission = query_expression()
+    environmentEndPoint = query_expression()
+
```

Comment:
We had a discussion on this in one of our weekly calls. It was decided that we would keep this class uniform with other classes in this file and it would be addressed by a database/table clean up that AWS would be undertaking separately. Should we address this after 2.2 in the clean up phase? If you think otherwise, I can take care of it in the next iteration.

Reply:
We will take it as part of the clean-up